tests/TypeProviderHelper.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+namespace WpOrg\Requests\Tests;
+
+use ArrayIterator;
+use EmptyIterator;
+use stdClass;
+use WpOrg\Requests\Tests\Fixtures\ArrayAccessibleObject;
+use WpOrg\Requests\Tests\Fixtures\StringableObject;
+
+final class TypeProviderHelper {
```

Comment:
This could use a piece of documentation, I guess.

Something like this, maybe?
```suggestion
/**
 * Helper class to provide an exhaustive list of types to test against type safety.
 */
final class TypeProviderHelper {
```